device/transport/mqtt/src/mqtt.ts

```diff
@@ -520,6 +520,17 @@ export class Mqtt extends EventEmitter implements DeviceTransport {
     });
   }
 
+  /**
+   * @private
+   * @method         module:azure-iot-device-mqtt:Mqtt#setModelId
+   * @description    This method sets the model ID so that it is included in the username on connect as per the Azure IoT Plug and Play requirements.
+   *
+   * @param {String}    modelId The model Id to set.
+   */
+  setModelId(modelId: string): void {
+    this._mid = '&model-id=' + modelId;
+  }
```

Comment:
There's already code in setOptions (line 550 below) to set model id.  Why add a second way do do this?

Reply:
A couple of reasons:
1. `setOptions` will eventually call `setOptions` on our base MQTT client which will overwrite any options passed in previously. My concern was that the user passes in an MQTT transport with some options that they already set and then we come in and internally call `setOptions` again and overwrite their options. 
2. `setOptions` takes a callback despite setting the modelId being synchronous. In the case of setting the model id, the callback is actually useless. We really shouldn't have a function that is sometimes synchronous and sometimes asynchronous. 

On second thought, I'm actually leaning toward removing this and using setOptions because:
1. Users shouldn't be using our transport objects directly to begin with. And this wouldn't be a breaking change because we would only call `setOptions` when the modelId is specified.
2. While it is bad practice to have a function that is sometimes synchronous and sometimes asynchronous, we already do it all over the place in our SDK. And perhaps it's a lesser evil that having two ways of doing the same thing. Whenever we get around to Track 2, we should be careful to avoid situations like this in the future.

Reply:
Fair enough.  I rememer all the pain that setOptions has caused in the past.  I did get burned by the unnecessary callback, and I dd get burned by the fact that it overwrites old options down in the transport.  I think you'll be fine either way you resolve this, and I hope we can eventually do better.

Reply:
Actually, I'll stick with the `setModelId` method but add some comments explaining the reason behind it. That'd be clearer, I think.